From 4c6c36dd9e3b29c37686b1f3c60a006a951fc78f Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@marvell.com>
Date: Fri, 12 Jul 2019 11:33:11 -0700
Subject: [PATCH 0939/1239] cmd: eth_cmds: Return usage for invalid argument
 values

Throw usage if invalid values are provided for set_fec and
set_phymod.

Change-Id: Icb249b6981dcaf7c8e9781af5ce05af6dd2323fc
Signed-off-by: Suneel Garapati <sgarapati@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/12491
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Suneel Garapati <sgarapati@caviumnetworks.com>
---
 cmd/marvell/eth_cmds.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmd/marvell/eth_cmds.c b/cmd/marvell/eth_cmds.c
index cce930e30b..79f45b84b8 100644
--- a/cmd/marvell/eth_cmds.c
+++ b/cmd/marvell/eth_cmds.c
@@ -67,7 +67,7 @@ static int do_ethparam_common(cmd_tbl_t *cmdtp, int flag, int argc,
 			return CMD_RET_FAILURE;
 		type = simple_strtol(argv[2], &endp, 0);
 		if (type < 0 || type > 2)
-			return CMD_RET_FAILURE;
+			return CMD_RET_USAGE;
 		ret = cgx_intf_set_fec(dev, type);
 	} else if (strcmp(cmd, "get_fec") == 0) {
 		ret = cgx_intf_get_fec(dev);
@@ -76,7 +76,7 @@ static int do_ethparam_common(cmd_tbl_t *cmdtp, int flag, int argc,
 			return CMD_RET_FAILURE;
 		type = simple_strtol(argv[2], &endp, 0);
 		if (type < 0 || type > 1)
-			return CMD_RET_FAILURE;
+			return CMD_RET_USAGE;
 		ret = cgx_intf_set_phy_mod_type(dev, type);
 	} else if (strcmp(cmd, "get_phymod") == 0) {
 		ret = cgx_intf_get_phy_mod_type(dev);
-- 
2.29.0

